gnucash/import-export/aqb/gnc-gwen-gui.c

```diff
@@ -949,11 +959,36 @@ get_input(GncGWENGui *gui, guint32 flags, const gchar *title,
     remember_pin_checkbutton = GTK_WIDGET(gtk_builder_get_object (builder, "remember_pin"));
     optical_challenge = GTK_IMAGE(gtk_builder_get_object (builder, "optical_challenge"));
     gtk_widget_set_visible(GTK_WIDGET(optical_challenge), FALSE);
+
+    flicker_challenge = GTK_WIDGET(gtk_builder_get_object(builder, "flicker_challenge"));
+    flicker_marker = GTK_WIDGET(gtk_builder_get_object(builder, "flicker_marker"));
+    flicker_hbox = GTK_WIDGET(gtk_builder_get_object(builder, "flicker_hbox"));
+    spin_barwidth = GTK_SPIN_BUTTON(gtk_builder_get_object(builder, "spin_barwidth"));
+    spin_delay = GTK_SPIN_BUTTON(gtk_builder_get_object(builder, "spin_delay"));
+
+    gtk_widget_set_visible(GTK_WIDGET(flicker_challenge), FALSE);
+    gtk_widget_set_visible(GTK_WIDGET(flicker_marker), FALSE);
+    gtk_widget_set_visible(GTK_WIDGET(flicker_hbox), FALSE);
+    gtk_widget_set_visible(GTK_WIDGET(spin_barwidth), FALSE);
+    gtk_widget_set_visible(GTK_WIDGET(spin_delay), FALSE);
+
     #ifdef AQBANKING6
     if(mimeType != NULL && pChallenge != NULL && lChallenge > 0)
     {
         gtk_widget_set_visible(GTK_WIDGET(optical_challenge), TRUE);
     }
+    else if (g_strcmp0(mimeType,"text/x-flickercode") == 0 && pChallenge != NULL)
```

Comment:
This is a bit obscure because it seems on the first four readings that it would never fire because the only reason the previous condition isn't true is that the flickercode block in `get_password_cb` doesn't set `lChallenge`. Reversing the order so that it's `if (g_strcmp0(mimeType, "text/x-flickercode") == 0... else if (mimeType != NULL...` would be clearer.